enterprise/server/cmd/ci_runner/main.go

```diff
@@ -855,6 +855,25 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace) error {
 			return nil
 		}
 	}
+	workspaceStatusEvent := &bespb.BuildEvent{
+		Id: &bespb.BuildEventId{Id: &bespb.BuildEventId_WorkspaceStatus{WorkspaceStatus: &bespb.BuildEventId_WorkspaceStatusId{}}},
+		Payload: &bespb.BuildEvent_WorkspaceStatus{WorkspaceStatus: &bespb.WorkspaceStatus{
+			Item: []*bespb.WorkspaceStatus_Item{
+				{Key: "BUILD_USER", Value: ar.username},
+				{Key: "BUILD_HOST", Value: ar.hostname},
+				{Key: "GIT_BRANCH", Value: *pushedBranch},
+				{Key: "GIT_TREE_STATUS", Value: "Clean"},
+				// Note: COMMIT_SHA may not actually reflect the current state of the
+				// repo since we merge the target branch before running the workflow;
+				// we set this for the purpose of reporting statuses to GitHub.
+				{Key: "COMMIT_SHA", Value: *commitSHA},
```

Comment:
One thing to note is that we sometimes set commitSha in sync() if it was not passed in as a flag (i.e. only a branch was passed). For that case, now the UI won't contain the commit sha

Reply:
If this becomes a major issue for people, we could write to the DB twice, once when seeing WorkflowConfigured and once when seeing WorkspaceStatus

Reply:
hmm actually I think this might affect any users who are currently using the ExecuteWorkflow API and only passing in a branch. For those people, we won't set COMMIT_SHA which would break GH status publishing.

@tylerwilliams wdyt about allowing workflows to do 1 extra invocation DB write - once when getting WorkflowConfigured, and once after checking out the git repo? The problem it'd solve is that today, we don't always know the commit_sha until checking out the repo, but if the git fetch is slow then the user is stuck in this "Invocation in progress..." page until it's done, rather than being able to see the status of the fetch